src/analysis/bounded_variation.lean

```diff
@@ -520,6 +530,96 @@ begin
   rw [← evariation_on.union f A B, ← inter_union_distrib_left, Icc_union_Icc_eq_Icc hab hbc],
 end
 
+/-TODO : where should the following two lemmas go? -/
+private lemma monotone_on_of_right_inv_on_of_maps_to_of_monotone_on
+  {α β : Type*} [partial_order α] [linear_order β] {φ : β → α} {ψ : α → β}
+  {t : set β} {s : set α} (φψs : right_inv_on ψ φ s) (ψts : maps_to ψ s t)
+  (hφ : monotone_on φ t) : monotone_on ψ s :=
+begin
+  rintro x xs y ys l,
+  rcases le_total (ψ x) (ψ y) with (ψxy|ψyx),
+  { exact ψxy, },
+  { cases le_antisymm l (φψs.eq ys ▸ φψs.eq xs ▸ hφ (ψts ys) (ψts xs) ψyx), refl, },
+end
+
+private lemma antitone_on_of_right_inv_on_of_maps_to_of_antitone_on
```

Comment:
can you deduce this one from the previous one by applying it to `order_dual α`, say?

Reply:
Thought it would be too much hassle, but let me try.

Reply:
turned out not to be a hassle at all!